controller/gateway/controller_reconciler_utils.go

```diff
@@ -780,6 +780,12 @@ func (g *gatewayConditionsAndListenersAwareT) setProgrammed() {
 			LastTransitionTime: metav1.Now(),
 		}
 		listenerStatus := listenerConditionsAware(listener)
+		rCond, ok := k8sutils.GetCondition(k8sutils.ConditionType(gatewayv1.ListenerConditionResolvedRefs), listenerStatus)
+		if ok && rCond.Status == metav1.ConditionFalse {
+			programmedCondition.Status = metav1.ConditionFalse
+			programmedCondition.Reason = string(gatewayv1.ListenerReasonPending)
+			programmedCondition.Message = "Listener references are not resolved yet."
+		}
```

Comment:
Why do we need to set such a condition here? We properly configure the `ResolvedRefs` conditions right after `setProgrammed`, in https://github.com/Kong/gateway-operator/blob/7f7db79dc821f212cc405b77fa7a4217fbf66ed6/controller/gateway/controller.go#L169-L170

Reply:
This doesn't set the `ResolvedRefs` condition but it checks it and based on that it sets the `Programmed`. This is to satisfy the conformance in https://github.com/kubernetes-sigs/gateway-api/blob/v1.1.0/conformance/tests/gateway-with-attached-routes.go#L96-L126.

I can see that GW API's `main` already [checks the `Accepted=False`](https://github.com/kubernetes-sigs/gateway-api/blob/8b7639171b5ae5007c8a7abd1796f02a053313cf/conformance/tests/gateway-with-attached-routes.go#L95-L139) as well but let's not get ahead of ourselves.

Reply:
oh I see, I didn't read carefully. Yeah let's move this one forward then :)